Purchases/SwiftSources/LocalReceiptParsing/DataConverters/UInt8+Extensions.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  UInt8+Extensions.swift
+//  Purchases
+//
+//  Created by Andrés Boedo on 7/24/20.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+extension UInt8 {
+    func bitAtIndex(_ index: UInt8) -> UInt8 {
+        guard 0 <= index && index <= 7 else { fatalError("invalid index: \(index)") }
+        let shifted = self >> (7 - index)
+        return shifted & 0b1
+    }
+
+    func valueInRange(from: UInt8, to: UInt8) -> UInt8 {
+        guard 0 <= from && from <= 7 else { fatalError("invalid index: \(from)") }
+        guard 0 <= to && to <= 7 else { fatalError("invalid index: \(to)") }
+        guard from <= to else { fatalError("from: \(from) can't be greater than to: \(to)") }
+
+        let range: UInt8 = to - from + 1
+        let shifted = self >> (7 - to)
+        let mask = maskForRange(range)
+        return shifted & mask
+    }
+}
+
+private extension UInt8 {
+    func maskForRange(_ range: UInt8) -> UInt8 {
+        guard 0 <= range && range <= 8 else { fatalError("range must be between 1 and 8") }
+        switch range {
```

Comment:
https://twitter.com/ctrlshifti/status/1288745146759000064?s=21 😂 
but I couldn't think of an alternative that was easier to understand at a glance, and it's just a few cases